charts/ccsm-helm/test/operate/deployment_test.go

```diff
@@ -0,0 +1,281 @@
+package operate
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+type deploymentTemplateTest struct {
+	suite.Suite
+	chartPath string
+	release   string
+	namespace string
+	templates []string
+}
+
+func TestDeploymentTemplate(t *testing.T) {
+	t.Parallel()
+
+	chartPath, err := filepath.Abs("../../")
+	require.NoError(t, err)
+
+	suite.Run(t, &deploymentTemplateTest{
+		chartPath: chartPath,
+		release:   "ccsm-helm-test",
+		namespace: "ccsm-helm-" + strings.ToLower(random.UniqueId()),
+		templates: []string{"charts/operate/templates/deployment.yaml"},
+	})
+}
+
+func (s *deploymentTemplateTest) TestContainerOverwriteImageTag() {
+	// given
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"operate.image.tag": "a.b.c",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", s.namespace),
+		ExtraArgs:      map[string][]string{"template": {"--debug"}, "install": {"--debug"}},
+	}
+
+	// when
+	output := helm.RenderTemplate(s.T(), options, s.chartPath, s.release, s.templates)
+	var deployment appsv1.Deployment
+	helm.UnmarshalK8SYaml(s.T(), output, &deployment)
+
+	// then
+	expectedContainerImage := "camunda/operate:a.b.c"
+	containers := deployment.Spec.Template.Spec.Containers
+	s.Require().Equal(1, len(containers))
```

Comment:
```suggestion
	s.Require().Len(containers, 1)
```

:wrench: Can be applied to others

Reply:
Hm with the `s.Require().Equal(1, len(containers))`

I get:

```
    deployment_test.go:79: 
        	Error Trace:	deployment_test.go:79
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestDeploymentTemplate/TestContainerOverwriteGlobalImageTag
    --- FAIL: TestDeploymentTemplate/TestContainerOverwriteGlobalImageTag (0.21s)
```

With `s.Require().Len(containers, 1)` I get:

```
    deployment_test.go:57: 
        	Error Trace:	deployment_test.go:57
        	Error:      	"[{operate camunda/operate:a.b.c [] []  [{http %!s(int32=0) %!s(int32=8080) TCP }] [] [] {map[cpu:{{%!s(int64=1000) %!s(resource.Scale=-3)} {%!s(*inf.Dec=<nil>)}  DecimalSI} memory:{{%!s(int64=2147483648) %!s(resource.Scale=0)} {%!s(*inf.Dec=<nil>)} 2Gi BinarySI}] map[cpu:{{%!s(int64=500) %!s(resource.Scale=-3)} {%!s(*inf.Dec=<nil>)} 500m DecimalSI} memory:{{%!s(int64=1073741824) %!s(resource.Scale=0)} {%!s(*inf.Dec=<nil>)} 1Gi BinarySI}]} [{config %!s(bool=false) /usr/local/operate/config/application.yml application.yml %!s(*v1.MountPropagationMode=<nil>) }] [] nil nil nil nil    nil %!s(bool=false) %!s(bool=false) %!s(bool=false)} {lol  [] []  [] [] [] {map[] map[]} [] [] nil nil nil nil    nil %!s(bool=false) %!s(bool=false) %!s(bool=false)}]" should have 1 item(s), but has 2
        	Test:       	TestDeploymentTemplate/TestContainerOverwriteImageTag
    --- FAIL: TestDeploymentTemplate/TestContainerOverwriteImageTag (0.16s)
```

I can add messages but :shrug: 

```
        	Messages:   	expect one container
    --- FAIL: TestDeploymentTemplate/TestContainerOverwriteImageTag (0.22s)
```